api/repo/repair.go

```diff
@@ -116,6 +116,35 @@ func RepairRepo(c *gin.Context) {
 		}
 	}
 
+	// get repo information from the source
+	sourceRepo, err := scm.FromContext(c).GetRepo(ctx, u, r)
+	if err != nil {
+		retErr := fmt.Errorf("unable to retrieve repo info for %s from source: %w", sourceRepo.GetFullName(), err)
+
+		util.HandleError(c, http.StatusBadRequest, retErr)
+
+		return
+	}
+	// if repo has a name change, then update DB with new name
+	// if repo has a org change, update org as well
+	if sourceRepo.GetName() != r.GetName() || sourceRepo.GetOrg() != r.GetOrg() {
+		r.SetPreviousName(r.GetName())
+		r.SetName(sourceRepo.GetName())
+		r.SetFullName(sourceRepo.GetFullName())
+		r.SetLink(sourceRepo.GetLink())
+		r.SetClone(sourceRepo.GetClone())
+		r.SetOrg(sourceRepo.GetOrg())
+		// send API call to update the repo
+		_, err := database.FromContext(c).UpdateRepo(ctx, r)
+		if err != nil {
+			retErr := fmt.Errorf("unable to rename repo %s to %s: %w", sourceRepo.GetFullName(), r.GetFullName(), err)
```

Comment:
wouldn't this error message always show the same full name? because of `r.SetFullName(sourceRepo.GetFullName())` a few lines above?